lib/bolt/transport/ssh.rb

```diff
@@ -187,10 +189,15 @@ def run_task(target, task, arguments, options = {})
         end
       end
 
-      def make_wrapper_stringio(task_path, stdin)
+      def make_wrapper_stringio(task_path, stdin, interpreter = nil)
         StringIO.new(<<-SCRIPT)
 #!/bin/sh
-'#{task_path}' <<'EOF'
+if test -z '#{interpreter}'; then
```

Comment:
It would probably be safer to put this conditional outside the wrapper script and just generate two different possible scripts depending on whether interpreter was passed in.

Reply:
Updated to implement the logic in ruby and simplified the wrapper script. Below is the previously proposed wrapper. 
```
      def make_wrapper_stringio(task_path, stdin, interpreter = nil)
        StringIO.new(<<-SCRIPT)
#!/bin/sh
if test -z '#{interpreter}'; then
  set -- '#{task_path}'
else
  set -- '#{interpreter}' '#{task_path}'
fi
"$@" <<'EOF'
#{stdin}
EOF
SCRIPT
      end
```